lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -3781,20 +3777,22 @@ swift::ModuleDecl *SwiftASTContext::GetModule(const SourceModule &module,
 
   // Report progress on module importing by using a callback function in
   // swift::ASTContext.
-  Progress progress("Importing Swift modules");
+  std::unique_ptr<Progress> progress;
   ast->SetPreModuleImportCallback(
       [&progress](llvm::StringRef module_name,
                   swift::ASTContext::ModuleImportKind kind) {
+        if (!progress)
+          progress = std::make_unique<Progress>("Importing Swift modules");
```

Comment:
Are these callbacks always called on the same thread?

Reply:
I'm actually unsure of this but I can check.

Reply:
I asked Adrian and he noted that while it's not guaranteed that the callbacks will be performed on the same thread, it is guaranteed that they will be called synchronously.